apache2/re.c

```diff
@@ -1970,6 +1970,48 @@ msre_ruleset *msre_ruleset_create(msre_engine *engine, apr_pool_t *mp) {
 
     return ruleset;
 }
+/**
+
+
+ * Creates a ruleset by copy the parent ruleset.
+ *
+ * @param mp apr pool structure
+ * @param parent_ruleset Parent's msre_ruleset
+ *
+ * return the address of created ruleset if OK.
+ * return NULL if Something went wrong.
+ */
+
+msre_ruleset *msre_ruleset_create_and_copy(msre_engine *engine, apr_pool_t *mp, msre_ruleset *parent_ruleset) {
+    msre_ruleset *child_ruleset;
+
+    child_ruleset = apr_pcalloc(mp, sizeof(msre_ruleset));
+    if (child_ruleset == NULL) return NULL;
+    child_ruleset->mp = mp;
+    child_ruleset->engine = engine;
+
+    if (parent_ruleset == NULL) {
```

Comment:
can this ever happen?

Reply:
Because the original code doesn't check the return code of copy_rules(), it works when the parent_ruleset is NULL. The original code take this condition into consideration. I think we should consider this corner case.